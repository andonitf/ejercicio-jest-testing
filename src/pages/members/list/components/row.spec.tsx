import * as React from 'react';
import { shallow } from 'enzyme';
import { Row } from './row';
import { Member } from '../viewModel';

describe('pages/members/list/components/row specs', () => {
    it('should render the component as expected when passing required properties', () => {
        // Arrange
        const member: Member = {
            id: 1,
            name: 'User Name 1',
            avatarUrl: 'http://fake-server-url/member/avatar/url/',
        };

        const props = { member };

        // Act
        const component = shallow(<Row member={props.member} />);

        // Assert
        expect(component).toMatchSnapshot();
    });
});
