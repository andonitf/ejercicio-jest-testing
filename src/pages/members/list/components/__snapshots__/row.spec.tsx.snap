// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components/row specs should render the component as expected when passing required properties 1`] = `
<tr>
  <td>
    <img
      src="http://fake-server-url/member/avatar/url/"
      style={
        Object {
          "maxWidth": "10rem",
        }
      }
    />
  </td>
  <td>
    <span>
      1
    </span>
  </td>
  <td>
    <span>
      User Name 1
    </span>
  </td>
</tr>
`;
